CMakeLists.txt

```diff
@@ -712,12 +722,21 @@ function(add_sanitizer sanitizer supported_compilers want_flag status_flag)
 	endif()
 endfunction()
 
-add_sanitizer(address "GNU|Clang|MSVC" WANT_DEBUG_ASAN STATUS_DEBUG_ASAN)
-add_sanitizer(thread "GNU|Clang" WANT_DEBUG_TSAN STATUS_DEBUG_TSAN)
-add_sanitizer(memory "Clang" WANT_DEBUG_MSAN STATUS_DEBUG_MSAN -fno-omit-frame-pointer)
+add_sanitizer(address "GNU|Clang|MSVC" WANT_DEBUG_ASAN STATUS_DEBUG_ASAN debug)
+add_sanitizer(thread "GNU|Clang" WANT_DEBUG_TSAN STATUS_DEBUG_TSAN debug)
+add_sanitizer(memory "Clang" WANT_DEBUG_MSAN STATUS_DEBUG_MSAN debug -fno-omit-frame-pointer)
+
 # UBSan does not link with vptr enabled due to a problem with references from PeakControllerEffect
 # not being found by PeakController
-add_sanitizer(undefined "GNU|Clang" WANT_DEBUG_UBSAN STATUS_DEBUG_UBSAN -fno-sanitize=vptr)
+add_sanitizer(undefined "GNU|Clang" WANT_DEBUG_UBSAN STATUS_DEBUG_UBSAN debug -fno-sanitize=vptr)
+
+add_sanitizer(realtime "Clang" WANT_RTSAN STATUS_RTSAN any)
+
+if(WANT_RTSAN)
+	add_compile_definitions(LMMS_NONBLOCKING=[[clang::nonblocking]])
+else()
+	add_compile_definitions(LMMS_NONBLOCKING=[[]])
+endif()
```

Comment:
I think this would be better in `lmmsconfig.h`. If you do it here, it's not clear in the code that `LMMS_NONBLOCKING` actually exists, so it may be undefined from the IDE's perspective and you may have to contend with red squiggly lines under functions that use it

Reply:
Fair concern, but I ran into issues doing it this way. My CMake knowledge is a bit janky so maybe I did something wrong. I'll try again.

Reply:
What's the motivation to be able to turn it off? I would have thought that an unknown attribute is just getting ignored, so if the compile flags for the sanitizer are off, it should still compile fine with `[[clang::nonblocking]]`?

Reply:
Ah, I didn't know unknown attributes were ignored. I thought it would cause a compiler error. I'll simplify this down.

Reply:
Not sure if we should even use the `LMMS_NONBLOCKING` macro then, but I do like how it doesn't make compiler specific references to the main codebase.